### PR TITLE
Add "clean" option support for pkgng module

### PR DESCRIPTION
**Pull request by dgeo**
##### ISSUE TYPE
  - Feature Pull Request

##### COMPONENT NAME
pkgng module

##### ANSIBLE VERSION
```
ansible 2.2.1.0
  config file = /home/dgeo/.ansible.cfg
  configured module search path = Default w/o overrides
```

##### SUMMARY
Runs `pkg clean` which removes cached package files. Possible values:
  * `none` (do not clean, default)
  * `obsolete` (clean only obsolete package files (e.g. old versions))
  * `all` (clean all package files)

Note this is a pure copy of @AMDmi3's original PR to
ansible-modules-extras#2339, adding @bcoca's suggestion : they should be credited here, I just act for this work not to be lost :)

### CODE REVIEW

File: lib/ansible/modules/packaging/os/pkgng.py [hunk 1]
```diff
@@ -86,6 +86,15 @@
         required: false
         choices: [ "yes", "no" ]
         default: no
+    clean:
+        version_added: "2.2"
```
Comment (nerzhul) [inline lib/ansible/modules/packaging/os/pkgng.py]: please set it to 2.3

**Review by nerzhul — CHANGES_REQUESTED** [review]
Nice addition, just a little thing to fix

**Comment by nerzhul** [comment]
needs_revision

**Comment by dgeo** [comment]
This time it should pass, removed `@`'s and version to 2.3.

**Comment by AMDmi3** [comment]
This is obsoleted by #27126, which fixes the conflicts and issue with incorrect default value for clean option.

**Comment by dgeo** [comment]
I'll let @AMDmi3 manage this in #27126 as he seems more up-to-date (and he made the original PR !) :)

**Pull request closed without merging**